### ISSUE TITLE
Question on Zero-shot Performance and Bounding Box Coordinates in Qwen-VL 2.5

### ISSUE THREAD

**Issue by LijunRio** [om-ai-lab/VLM-R1#288]
Hi,

Thank you so much for your work :)

I have a question regarding the zero-shot performance of the Qwen-VL 2.5 series. As reported both in the paper and on your blog, the zero-shot performance seems relatively low. 

I noticed in the released codebase from your repo, the predicted bounding boxes are in raw pixel values — for example, the model outputs coordinates like the following, which can be directly used to draw rectangles on the image:

```json
[
  {"bbox_2d": [105, 301, 232, 480], "label": "person"}, 
  {"bbox_2d": [210, 263, 290, 476], "label": "person"}, 
  {"bbox_2d": [153, 265, 204, 338], "label": "person"}, 
  {"bbox_2d": [83, 274, 165, 452], "label": "person"}
]
```

However, in the Qwen-VL 2.5 official repo it is mentioned that due to specific image transformations, **the bounding box outputs from the model should be rescaled**— i.e., normalized values need to be converted to pixel coordinates using something like `(bbox / input_length) * image_width` (or height).  https://github.com/QwenLM/Qwen2.5-VL/blob/main/cookbooks/spatial_understanding.ipynb

![Image](https://github.com/user-attachments/assets/eab2c6bf-d19f-41fc-9f34-af4235da8154)

Could this discrepancy be part of the reason why the zero-shot performance reported here is lower than what is shown in the original Qwen-VL 2.5 technical report?

Also, does this affect training? From the code, it looks like the training annotations are also already in pixel coordinates.

Looking forward to your reply!

Best regards,
Jun

**Comment by Rakshith12-pixel** [comment]
@LijunRio Do you have a training code to train qwen for bounding box detection for a specific  task??
Thanks in advance
